File: Q3_Cpp/src/Product/Electronic/Electronic.cpp
#include "../../../include/Product/Electronic/Electronic.h"

// Constructor for Electronic
Electronic::Electronic(std::string id, std::string name, Supplier supplier, int packagingUnits, double price, std::string manufacturer, std::string model)
    : Product(id, name, supplier, packagingUnits, price), manufacturer(manufacturer), model(model) {
}

// Getter and setter for manufacturer
std::string Electronic::getManufacturer() const {
    return manufacturer;
}

void Electronic::setManufacturer(const std::string& manufacturer) {
    this->manufacturer = manufacturer;
}

// Getter and setter for model
std::string Electronic::getModel() const {
    return model;
}

void Electronic::setModel(const std::string& model) {
    this->model = model;
}

File: Q1_Java/src/main/java/Product/Electronic/Electronic.java
package Product.Electronic; // Package for Electronic products

import Product.Product;
import Product.Supplier;

// Abstract Electronic class, extends Product
public abstract class Electronic extends Product {
    private String manufacturer; // Manufacturer of the electronic product
    private String model; // Model of the electronic product

    // Constructor for Electronic
    public Electronic(String id, String name, Supplier supplier, int packagingUnits, double price, String manufacturer, String model) {
        super(id, name, supplier, packagingUnits, price); // Calling Product's constructor
        this.manufacturer = manufacturer; // Setting manufacturer
        this.model = model; // Setting model
    }

    // Getter and setter for manufacturer
    public String getManufacturer() { return manufacturer; }
    public void setManufacturer(String manufacturer) { this.manufacturer = manufacturer; }

    // Getter and setter for model
    public String getModel() { return model; }
    public void setModel(String model) { this.model = model; }
}
